app/controllers/recipes_controller.rb

```diff
@@ -0,0 +1,87 @@
+class RecipesController < ApplicationController
+  before_action :set_user_recipes, only: %i[show update destroy]
+  before_action :set_user_recipe, only: %i[show update]
+
+  def index
+    @recipes = current_user.recipes
+  end
+
+  def public_recipes
+    @recipes = Recipe.where("public = 'true' ")
+  end
```

Comment:
- [ ] You should avoid making SQL queries from the controller. Your controllers should only redirect data that they get from models. You should always make SQL queries from the model.